src/assets/scripts/airport/Route/RouteModel.js

```diff
@@ -0,0 +1,134 @@
+import _uniqueId from 'lodash/uniqueId';
+
+/**
+ * Symbol that divides each route segment
+ *
+ * @property SEGMENT_SEPERATION_SYMBOL
+ * @type {string}
+ * @final
+ */
+const SEGMENT_SEPERATION_SYMBOL = '.';
+
+/**
+ * @class RouteModel
+ */
+export default class RouteModel {
+    /**
+     * 'BETHL.GRNPA1.KLAS'
+     *
+     * @for RouteModel
+     * @constructor
+     * @param routeString {string}
+     */
+    constructor(routeString) {
+        if (typeof routeString === 'undefined' || typeof routeString !== 'string') {
+            return;
+        }
+
+        if (!this._isValidRouteString(routeString)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Invalid routeString passed to RouteModel. Expected a routeString of the shape ORIGIN.BASE.DESTINATION but instead received ${routeString}`);
+        }
+
+        /**
+         * @property
+         * @type {string}
+         */
+        this._id = _uniqueId();
+
+        /**
+         * @property origin
+         * @type {string}
+         * @default ''
+         */
+        this.origin = '';
+
+        /**
+         * @property base
+         * @type {string}
+         * @default ''
+         */
+        this.base = '';
+
+        /**
+         * @property destination
+         * @type {string}
+         * @default ''
+         */
+        this.destination = '';
+
+        return this._init(routeString);
+    }
+
+    /**
+     * A single string that represents the entire route
+     *
+     * @property routeString
+     * @return {string}
+     */
+    get routeString() {
+        return `${this.origin}.${this.base}.${this.destination}`;
```

Comment:
Thank you.

In its current implementation, yes, it will work all the time.  I know this because I have tests that prove it.  But...

Now that I have begun working through the fms, `Leg` and `Waypoint` models, I can see that this class will need some work.  Thankfully, the foundation is there.  So we can say goodbye to:
```javascript
const entry = this.route.split('.')[0];
const star = this.route.split('.')[1];
const apt = this.route.split('.')[2];
```

It now reads better and is encapsulated in a central spot.